File: backend/entity/foodchoice_disease.go
package entity

import (
	"gorm.io/gorm"
)

type FoodchoiceDisease struct {
	gorm.Model
	Description string

	// Diseaseทำหน้าที่เป็น FK
	DiseaseID uint
	Disease Disease `gorm:"foreignKey:DiseaseID"`

	// FoodChoice ทำหน้าที่เป็น FK
	FoodChoiceID uint
	FoodChoice FoodChoice `gorm:"foreignKey:FoodChoiceID"`

}

File: frontend/src/interfaces/FoodchoiceDisease.ts
/* export interface FoodchoiceDiseaseInterface {
    ID?: number;
    Description?: string;
    DiseaseID?: number; //FK
    FoodChoiceID?: number; //FK       
} */

export interface FoodchoiceDiseaseInterface {
  ID?: number;
  DiseaseID: number;
  FoodChoiceID: number;
  Description: string;
  FoodChoice?: {
    ID: number;
    FoodName: string;
  };
  Disease?: {
    ID: number;
    Name: string;
    Stage: string;
  };
}